staging/src/k8s.io/dynamic-resource-allocation/api/types.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
```

Comment:
I really don't like the duplication of this API.

If someone updates the api in staging, it seems to be uncommon to know that these file must also be manually updated.

Reply:
Code generation will fail when they get out of sync by accident (deep copy).

Besides, if someone touches parts of the ResourceSlice API, they have to modify `staging/src/k8s.io/dynamic-resource-allocation/structured` and should better know how this works.

Reply:
Maybe we could call this package (api) something more fitting for generated code? Reviewing this last night, I didn't really see anything that said this was generated code.